packages/openapi-ts/src/plugins/@pinia-colada/sdk/types.d.ts

```diff
@@ -0,0 +1,51 @@
+import type { IR } from '../../../ir/types';
+import type { Plugin } from '../../types';
+
+export interface QueryHooks {
+  /**
+   * Customize the query key for a specific operation
+   */
+  getQueryKey?: (operation: IR.OperationObject) => string[];
+
+  /**
+   * Override the default query detection logic for an operation.
+   * Return true to force treat as query, false to force treat as mutation.
+   * Return undefined to use default detection logic.
+   */
+  isQuery?: (operation: IR.OperationObject) => boolean | undefined;
+
+  /**
+   * Called when a mutation is created. Return false to skip generating this mutation.
+   */
+  onMutation?: (operation: IR.OperationObject) => boolean | void;
+
+  /**
+   * Called when a query is created. Return false to skip generating this query.
+   */
+  onQuery?: (operation: IR.OperationObject) => boolean | void;
+}
+
+export interface Config extends Plugin.Name<'@pinia-colada/sdk'> {
+  /**
+   * Export queries/mutations from index?
+   * @default true
+   */
+  exportFromIndex?: boolean;
+
+  /**
+   * Group queries by tag into separate files?
+   * @default false
+   */
+  groupByTag?: boolean;
+
+  /**
+   * Hooks to customize query/mutation generation
+   */
+  hooks?: QueryHooks;
```

Comment:
Same question here as above

Reply:
Lessons learned from trying to get things working with the old library, having even just some basic hooks available makes things a hundred times easier if you some funky endpoints and need to filter/exclude or translate some properties to something the this lib will understand (sometimes you just have one endpoint you need to edit on-read), it's not always canonical OpenAPI stuff that a PR could be made for. (Just trying to get a bunch people to export OpenAPI specs when they are close to retirement and don't want to deal with something new is enough without trying to convince them to adhere to standards for things are supposed work or be labeled 😅🙃)

Reply:
Same reply as above. We will most likely want to offer some consistency with these APIs as this requirement comes up with other plugins too and I've been purposefully avoiding it for now

Reply:
@josh-hemphill I'd prefer to flatten these options so instead of `plugin.hooks.getQueryKey()` it would be simply `plugin.getQueryKey()`. I'm not ready to define what a Hooks API is and this approach would remove that problem. I am working on other plugins and facing the same questions there. Thoughts?

Reply:
I thought I was removing the hooks?

Reply:
There's no need since they're clearly useful for you! And I'm sure sooner or later someone else would want to customise these options too.

I will have a separate question about `getQueryKey()` later, but will wait for snapshots before going there

Reply:
The `on*` ones seem fine, but I'm thinking renaming the others to `resloveQ*` or something